File: src/types/event.ts
export interface CalendarEvent {
  id: string;
  title: string;
  start: string; // YYYY-MM-DD HH:mm
  end: string;   // YYYY-MM-DD HH:mm
  description?: string;
  reminderMinutesBefore?: number;
}

export type NewCalendarEvent = Omit<CalendarEvent, 'id'> & { id?: string };

File: flutter_app/lib/models/event.dart
class CalendarEvent {
  final String id;
  final String title;
  final DateTime start;
  final DateTime end;
  final String? location;
  final String? notes;

  CalendarEvent({
    required this.id,
    required this.title,
    required this.start,
    required this.end,
    this.location,
    this.notes,
  });
}


